networkit/cpp/centrality/GroupClosenessLocalSwaps.cpp

```diff
@@ -0,0 +1,362 @@
+// networkit-format
+
+#include <algorithm>
+#include <cassert>
+#include <omp.h>
+#include <queue>
+
+#include <networkit/auxiliary/Random.hpp>
+#include <networkit/centrality/GroupClosenessLocalSwaps.hpp>
+
+namespace NetworKit {
+
+GroupClosenessLocalSwaps::GroupClosenessLocalSwaps(const Graph &G, const std::vector<node> &group,
+                                                   count maxSwaps)
+    : GroupClosenessLocalSwaps(G, group.begin(), group.end(), maxSwaps) {}
+
+void GroupClosenessLocalSwaps::init() {
+    const auto n = G->upperNodeIdBound();
+
+    distance.assign(n, 0);
+    gamma.assign(n * group.size(), false);
+    canSwap.assign(group.size(), false);
+    visited.assign(n, false);
+
+    idxMap.clear();
+    idxMap.reserve(group.size());
+
+    stack.assign(n, 0);
+    farness.assign(group.size(), 0);
+    farnessDecrement.assign(group.size(), 0);
+    sumOfMins.assign(n, 0);
+
+    intDistributions.resize(omp_get_max_threads());
+    urngs.reserve(omp_get_max_threads());
+    for (index i = 0; i < static_cast<index>(omp_get_max_threads()); ++i)
+        urngs.emplace_back(Aux::Random::getURNG());
+
+    for (size_t i = 0; i < group.size(); ++i) {
+        const auto u = group[i];
+        idxMap[u] = i;
+        gamma[u * group.size() + i] = 1;
+    }
+
+#ifdef __AVX2__
+    randVec.resize(n);
+#else
+    randVec.resize(K * n);
+#endif // __AVX2__
+
+    totalSwaps = 0;
+    hasRun = false;
+}
+
+void GroupClosenessLocalSwaps::run() {
+    init();
+
+    while (findAndSwap() && ++totalSwaps < maxSwaps) {
+    }
```

Comment:
This loops strange. IMHO the following is cleaner:
```suggestion
    while (findAndSwap() && ++totalSwaps < maxSwaps)
        ; // Just iterate.
```